src/datepicker/datepicker-keyboard-service.ts

```diff
@@ -10,18 +10,22 @@ import {Key} from '../util/key';
  */
 @Injectable({providedIn: 'root'})
 export class NgbDatepickerKeyboardService {
+  constructor(private _calendar: NgbCalendar) {}
```

Comment:
As `KeyboardService` provided in `root`, I wonder what will get injected when the `calendar` is overridden in the datepicker level ?

I guess it should be left as an argument to the `processKey` function